google/cloud/storage/oauth2/google_credentials_test.cc

```diff
@@ -215,11 +215,31 @@ TEST_F(GoogleCredentialsTest, LoadValidServiceAccountCredentialsFromFilename) {
   EXPECT_EQ(typeid(*ptr), typeid(ServiceAccountCredentials<>));
 }
 
+TEST_F(GoogleCredentialsTest,
+       LoadValidServiceAccountCredentialsFromFilenameWithOptionalArgs) {
+  std::string filename = ::testing::TempDir() + SERVICE_ACCOUNT_CRED_FILENAME;
+  SetupServiceAccountCredentialsFileForTest(filename);
+
+  // Test that the service account credentials are loaded from a file.
+  auto creds = CreateServiceAccountCredentialsFromJsonFilePath(
+      filename,
+      std::set<std::string>{
+          "https://www.googleapis.com/auth/devstorage.full_control"},
+      "user@foo.bar");
+  ASSERT_TRUE(creds.ok()) << "status=" << creds.status();
+  auto credentials = std::move(*creds);
+  auto ptr = credentials.get();
```

Comment:
No need to move out of the StatusOr. Instead use it like a smart pointer. Like
```
auto* ptr = creds->get();
```

Reply:
Good catch. I think this was an artifact of tacking on Status/StatusOr stuff to existing code :)